discord-scripts/invite-management.ts

```diff
@@ -75,67 +128,183 @@ export default async function sendInvite(discordClient: Client, robot: Robot) {
-              name: roleName,
-              reason: `Role for ${channel.name} channel`,
-            })
+      if (
+        !interaction.guild ||
+        !(interaction.channel instanceof TextChannel) ||
+        (interaction.channel.parent &&
+          interaction.channel.parent.name !== "defense")
+      ) {
+        await interaction.reply({
+          content:
+            "This command can only be run in channels under the 'defense' category.",
+          ephemeral: true,
+        })
+        return
+      }
 
-            await channel.permissionOverwrites.create(role, {
-              ViewChannel: true,
+      const clientName = interaction.options.get("client-name")
+      if (!clientName) {
+        await interaction.reply({
+          content: "Client name is required for the defense-audit command.",
+          ephemeral: true,
+        })
+        return
+      }
+
+      try {
+        if (typeof clientName.value === "string") {
+          await interaction.deferReply({ ephemeral: true })
+          const normalizedClientName = clientName.value
+            .replace(/\s+/g, "-")
+            .toLowerCase()
+          const internalChannelName = `int-${normalizedClientName}-audit`
+          const externalChannelName = `ext-${normalizedClientName}-audit`
+
+          const defenseCategory = interaction.guild.channels.cache.find(
+            (c) => c.name === "defense",
+          )
+
+          if (!defenseCategory) {
+            await interaction.reply({
+              content: "Defense category does not exist.",
+              ephemeral: true,
             })
-            channel.send(
-              `**${role.name}** role created and permissions set for **${channel.name}**`,
-            )
-            robot.logger.info(
-              `${role.name} role created and permissions set for channel ${channel.name}`,
-            )
-          } else {
-            robot.logger.info(
-              `Skipping role creation due to empty client name for channel ${channel.name}`,
+            return
+          }
+
+          const internalChannel = await interaction.guild.channels.create({
+            name: internalChannelName,
+            type: ChannelType.GuildText,
+            parent: defenseCategory.id,
+          })
+
+          const externalChannel = await interaction.guild.channels.create({
+            name: externalChannelName,
+            type: ChannelType.GuildText,
+            parent: defenseCategory.id,
+          })
+
+          const internalRoleName = `Defense Internal: ${clientName.value}`
+          const externalRoleName = `Defense External: ${clientName.value}`
+          const internalRole = await interaction.guild.roles.create({
+            name: internalRoleName,
+            reason: "Role for internal audit channel",
+          })
+          const externalRole = await interaction.guild.roles.create({
+            name: externalRoleName,
+            reason: "Role for external audit channel",
+          })
```

Comment:
Same deal here—though maybe `.create` on the Discord API already does this?

Reply:
Switched this one around as it seems like actually discord `.create` does not, it will make an ever infinite amount of roles with the same name! Also the case for channel names, I am seeing 😆